### ISSUE TITLE
conflict with installed nlohman/json

### ISSUE THREAD

**Issue by KrisThielemans** [root-project/root#14634]
### Check duplicate issues.

- [ ] Checked for duplicates

### Description

I have a compilation error in the case with an existing version of the nlohmann library
```
/home/runner/install/include/nlohmann/json.hpp:3435:8: error: redefinition of default argument for ‘class T’
 3435 | struct adl_serializer;
      |        ^~~~~~~~~~~~~~
In file included from /home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/ROOT/graf3d/eve7/inc/ROOT/REveShape.hxx:15,
                 from /home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/ROOT/graf3d/eve7/inc/ROOT/REveBox.hxx:15,
                 from /home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/ROOT/graf3d/eve7/src/REveBox.cxx:12:
/home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/ROOT/graf3d/eve7/inc/ROOT/REveElement.hxx:39:10: note: original definition appeared here
   39 | template<typename T = void, typename SFINAE = void>
      |          ^~~~~~~~
In file included from /home/runner/work/SIRF-SuperBuild/SIRF-SuperBuild/build/sources/ROOT/graf3d/eve7/src/REveBox.cxx:18:
/home/runner/install/include/nlohmann/json.hpp:3448:7: error: redefinition of default argument for ‘template<class U, class V, class ... Args> class ObjectType’
 3448 | class basic_json;
```

I believe this is cause by the fact that the installed version is v3.10.4 (which doesn't have `json_fwd.h`). Relevant bits of code:
- CMake code [checks for version](https://github.com/root-project/root/blob/f98afaed8286ae156d364bedd36c440bc22c0875/cmake/modules/SearchInstalledSoftware.cmake#L87)
- CMake [defaults to "build-in" OFF](https://github.com/root-project/root/blob/f98afaed8286ae156d364bedd36c440bc22c0875/cmake/modules/RootBuildOptions.cmake#L100)
- `REveElement.hxx` contains a work-around if `json_fwd.h` is not found, i.e. it [defines its own](https://github.com/root-project/root/blob/25e81912b499df1703bc6bed47623cfbbce036f7/graf3d/eve7/inc/ROOT/REveElement.hxx#L25-L37).

I believe the latter is causing the problem: somehow the existing (old) `json.h` has been read, and then creates a conflict with this definition.

I think this work-around needs to be removed. Possibly the easiest is to just require a more recent version.

### Reproducer

install old version and try to build ROOT

### ROOT version

I tried with v6-28-12 and 3-13.-3 with gcc-11.

### Installation method

compile from source

### Operating system

Ubuntu 22.04

### Additional context

There is no problem with v6-26-10

**Comment by bellenot** [comment]
@linev might be interested, he fixed a couple of issue with `nlohman/json` in the past

**Comment by linev** [comment]
Just to understand - is ROOT compiled with external or builtin `nlohmann/json.hpp`?

Strange that `v3.10.4` makes problem. 
ROOT builtin version is `3.10.5` and definitions in `REveElement.hxx` should be compatible with it.

**Comment by KrisThielemans** [comment]
> Just to understand - is ROOT compiled with external or builtin `nlohmann/json.hpp`?
> 
I used default options, so `builtin_nlohmannjson=OFF`

> Strange that `v3.10.4` makes problem. ROOT builtin version is `3.10.5` and definitions in `REveElement.hxx` should be compatible with it.

yes, that's strange. I haven't checked the diff between 3.10.5 and 3.10.4.

**Comment by linev** [comment]
And seems to be, starting from llvm13 and root 6.30 use of the `json_fwd.h` is no longer necessary.
Before interpreter always had problems to correctly parse `nlohmann/json.hpp`

I will do several more checks and probably just remove it's usage from master and 6.30 branches.

**Comment by linev** [comment]
I remove usage of `nlohmann/json_fwd.hpp` in 6.30 branch. 
Fix will be release with 6.30.06 version
